eng/illink.targets

```diff
@@ -105,8 +103,42 @@
     </ItemGroup>
   </Target>
 
+  <PropertyGroup>
+    <ILLinkResourcesSubstitutionIntermediatePath>$(IntermediateOutputPath)ILLink.Resources.Substitutions.xml</ILLinkResourcesSubstitutionIntermediatePath>
+    <GenerateResourcesSubstitutions Condition="'$(GenerateResourcesSubstitutions)' == '' and '$(StringResourcesPath)' != ''">true</GenerateResourcesSubstitutions>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(GenerateResourcesSubstitutions)' == 'true'">
+    <ILLinkSubstitutionsXmls Include="$(ILLinkResourcesSubstitutionIntermediatePath)" />
+  </ItemGroup>
+  
+  <!-- If a library uses string resources, the following target generates a substitution xml that will be embedded on the
+  library so that if a consumer wants to run the linker they can specify a feature switch to strip out all resources
+  from the assembly. -->
+  <Target Name="GenerateResourcesSubstitutionFile"  
+          Condition="'$(GenerateResourcesSubstitutions)' == 'true'"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(ILLinkResourcesSubstitutionIntermediatePath)">
+
+    <PropertyGroup>
+      <ILLinkResourcesSubstitutionTemplate>$(MSBuildThisFileDirectory)ILLink.Substitutions.Resources.template</ILLinkResourcesSubstitutionTemplate>
+    </PropertyGroup>
+
+    <WriteLinesToFile File="$(ILLinkResourcesSubstitutionIntermediatePath)"
+                      Lines="$([System.IO.File]::ReadAllText('$(ILLinkResourcesSubstitutionTemplate)')
+                                                 .Replace('{AssemblyName}', '$(AssemblyName)')
+                                                 .Replace('{StringResourcesName}', '$(StringResourcesName)'))"
+                      Overwrite="true" />
+
+    <ItemGroup>
+      <FileWrites Include="$(ILLinkResourcesSubstitutionIntermediatePath)" />
+    </ItemGroup>
+    <Message Importance="Low" Text="Generated $(ILLinkResourcesSubstitutionIntermediatePath) substitution file." />
```

Comment:
Do we need this?

Reply:
It only really gets displayed in diagnostic logs and it is how we do it in other targets that generate files although technically now that we do most of our diagnosis on binlogs the answer would be that it is not really needed. If we don't have anything against keeping it, I'll do that just to mimic what we do for other generated files (like assemblyInfo and other files that get generated under the obj directory)

Reply:
You don't need to change it here, but IMO this feels like cruft to me. It doesn't provide value.